language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -339,6 +352,35 @@ private static Hover getFunctionHoverMarkupContent(FunctionSymbol symbol, HoverC
                     returnTypeName + " : " + documentation.get().returnDescription().get();
             hoverContent.add(returnDoc);
         }
+        if (node.kind() == SyntaxKind.CAPTURE_BINDING_PATTERN
+                && symbol.typeDescriptor().returnTypeDescriptor().isPresent()) {
+            String returnTypeOfAnonFunc = symbol.typeDescriptor().returnTypeDescriptor().get().typeKind().getName();
+            String varName = symbol.getName().isPresent() ? " " + symbol.getName().get() : "";
+            String modifiedVariable = quotedString(returnTypeOfAnonFunc) + varName;
+            hoverContent.add(modifiedVariable);
+        }
```

Comment:
Any specific reason to check for the node kind? Also, I feel like the `CAPTURE_BINDING_PATTERN` check is too abstract.

Reply:
Node kind was checked to differentiate a variable in an `Anonymous Function Expression` with a `Function Definition Node`. But with the [commit](https://github.com/ballerina-platform/ballerina-lang/pull/32622/commits/6ca0e5099914c62e945375891107918012876a7b), this `nodeAtCursor` implementation was removed hence this code snippet as well.